eternalcore-core/src/main/java/com/eternalcode/core/feature/rtp/RandomTeleportService.java

```diff
@@ -0,0 +1,120 @@
+package com.eternalcode.core.feature.rtp;
+
+import com.eternalcode.core.configuration.implementation.PluginConfiguration;
+import com.eternalcode.core.scheduler.Scheduler;
+import io.papermc.lib.PaperLib;
+import org.bukkit.Chunk;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Player;
+
+import java.util.EnumSet;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+public class RandomTeleportService {
+
+    private static final Set<Material> UNSAFE_BLOCKS = EnumSet.of(
+        Material.LAVA,
+        Material.WATER,
+        Material.CACTUS,
+        Material.FIRE,
+        Material.COBWEB,
+        Material.SWEET_BERRY_BUSH,
+        Material.MAGMA_BLOCK
+    );
+
+    private static final Set<Material> AIR_BLOCKS = EnumSet.of(
+        Material.AIR,
+        Material.CAVE_AIR,
+        Material.VOID_AIR,
+        Material.GRASS,
+        Material.TALL_GRASS,
+        Material.VINE
+    );
+
+    private static final int DEFAULT_WORLD_BORDER_SIZE = 29_999_984;
+
+    private final Scheduler scheduler;
+
+    private final PluginConfiguration.Teleport teleportConfiguration;
+
+    private final Random random = new Random();
+
+    public RandomTeleportService(PluginConfiguration.Teleport teleportConfiguration, Scheduler scheduler) {
+        this.teleportConfiguration = teleportConfiguration;
+        this.scheduler = scheduler;
+    }
+
+    public CompletableFuture<TeleportResult> teleport(Player player) {
+        return this.getSafeRandomLocation(player.getWorld(), 100)
```

Comment:
Próby lepiej dać na 10 - 15 bo gracz będzie długo czekać. Już lepiej szybko jemu to zaresponsować